src/java/htsjdk/variant/vcf/VCFHeader.java

```diff
@@ -273,29 +272,36 @@ public void loadVCFVersion() {
-            } else if ( line instanceof VCFFormatHeaderLine ) {
-                final VCFFormatHeaderLine formatLine = (VCFFormatHeaderLine)line;
-                addMetaDataMapBinding(mFormatMetaData, formatLine);
-            } else if ( line instanceof VCFFilterHeaderLine ) {
-                final VCFFilterHeaderLine filterLine = (VCFFilterHeaderLine)line;
-                mFilterMetaData.put(filterLine.getID(), filterLine);
-            } else if ( line instanceof VCFContigHeaderLine ) {
-                contigMetaData.add((VCFContigHeaderLine)line);
-            } else {
-                mOtherMetaData.put(line.getKey(), line);
-            }
+            addMetadataHeaderLine(line);
         }
+    }
 
-        if ( hasFormatLine(VCFConstants.GENOTYPE_LIKELIHOODS_KEY) && ! hasFormatLine(VCFConstants.GENOTYPE_PL_KEY) ) {
-            if ( GeneralUtils.DEBUG_MODE_ENABLED ) {
-                System.err.println("Found " + VCFConstants.GENOTYPE_LIKELIHOODS_KEY + " format, but no "
-                                   + VCFConstants.GENOTYPE_PL_KEY + " field.  We now only manage PL fields internally"
-                                   + " automatically adding a corresponding PL field to your VCF header");
+    /**
+     * add a single header line to the appropriate map/list depending on its type
+     */
+    private void addMetadataHeaderLine(final VCFHeaderLine line) {
+        if ( line instanceof VCFInfoHeaderLine )  {
+            final VCFInfoHeaderLine infoLine = (VCFInfoHeaderLine)line;
+            addMetaDataMapBinding(mInfoMetaData, infoLine);
+        } else if ( line instanceof VCFFormatHeaderLine ) {
+            final VCFFormatHeaderLine formatLine = (VCFFormatHeaderLine)line;
+            addMetaDataMapBinding(mFormatMetaData, formatLine);
+
+            if ( hasFormatLine(VCFConstants.GENOTYPE_LIKELIHOODS_KEY) && ! hasFormatLine(VCFConstants.GENOTYPE_PL_KEY) ) {
+                if ( GeneralUtils.DEBUG_MODE_ENABLED ) {
+                    System.err.println("Found " + VCFConstants.GENOTYPE_LIKELIHOODS_KEY + " format, but no "
+                            + VCFConstants.GENOTYPE_PL_KEY + " field.  We now only manage PL fields internally"
+                            + " automatically adding a corresponding PL field to your VCF header");
+                }
+                addMetaDataLine(new VCFFormatHeaderLine(VCFConstants.GENOTYPE_PL_KEY, VCFHeaderLineCount.G, VCFHeaderLineType.Integer, "Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification"));
             }
-            addMetaDataLine(new VCFFormatHeaderLine(VCFConstants.GENOTYPE_PL_KEY, VCFHeaderLineCount.G, VCFHeaderLineType.Integer, "Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification"));
+        } else if ( line instanceof VCFFilterHeaderLine ) {
+            final VCFFilterHeaderLine filterLine = (VCFFilterHeaderLine)line;
+            mFilterMetaData.put(filterLine.getID(), filterLine);
```

Comment:
It's a bit inconsistent that for info and format lines, we call `addMetaDataMapBinding()` and therefore preserve the existing value when adding a line with an existing ID, whereas for filter and "other" lines we always replace the existing value in the map with new values having the same ID (since for those lines you call `Map.put()` directly instead of `addMetaDataMapBinding()`). Can you think of a good reason for this inconsistency? Perhaps we should make the behavior consistent here -- either always replace existing values with the same key, or never replace them.